python/cudf/cudf/core/column/string.py

```diff
@@ -4492,22 +4492,17 @@ def base_size(self):
                 / self.base_children[0].dtype.itemsize
             )
 
-    def sum(self, dtype=None):
-        return self.str().cat()
-
-    def product(self, dtype=None):
-        raise TypeError("can't multiply sequence by non-int of type 'object'")
-
-    def mean(self, dtype=np.float64):
-        raise NotImplementedError(
-            "mean for Series of type 'object' is not yet implemented."
+    def sum(self, skipna=None, dtype=None, min_count=0):
+        result_col = self._process_for_reduction(
+            skipna=skipna, min_count=min_count
         )
+        if isinstance(result_col, cudf.core.column.ColumnBase):
+            return result_col.str().cat()
+        else:
+            return result_col
 
-    def var(self, ddof=1, dtype=np.float64):
-        raise TypeError("unsupported operation for object of type 'object'")
-
-    def std(self, ddof=1, dtype=np.float64):
-        raise TypeError("unsupported operation for object of type 'object'")
+    def product(self, skipna=None, dtype=None, min_count=0):
+        raise TypeError("can't multiply sequence by non-int of type 'object'")
```

Comment:
This error isn't really clear even if it matches Pandas... Could we improve this?

Reply:
Removed this special error from `StringColumn.product` as `ColumnBase.product` is having a clear error now.